modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -70,10 +83,32 @@ public void setParameters(Map<String, Object> parameters) {
    * @return a {@link ResolverSet}
    * @throws {@link ConfigurationException} if the exclusiveness condition between parameters is not honored
    */
-  protected ResolverSet getParametersAsResolverSet(EnrichableModel model) throws ConfigurationException {
+  protected ResolverSet getParametersAsResolverSet(ParameterizedModel model, Predicate<ParameterModel>... predicates)
+      throws ConfigurationException {
+
     ResolverSet resolverSet = new ResolverSet();
-    getParameters().forEach((key, value) -> resolverSet.add(key, toValueResolver(value)));
-    checkParameterGroupExclusivenessForModel(model, getParameters().keySet());
+    Map<String, Object> parameters = getParameters();
+    model.getParameterModels().stream()
+        .filter(CompositePredicate.of(predicates))
+        .forEach(p -> {
+          String parameterName = getMemberName(p, p.getName());
+          ValueResolver<?> resolver = null;
+          if (parameters.containsKey(parameterName)) {
+            resolver = toValueResolver(parameters.get(parameterName));
+          } else if (isNullSafe(p)) {
+            resolver = NullSafeValueResolverWrapper
+                .of(p.getType(), new StaticValueResolver<>(null), p.getModelProperties(), muleContext);
+          }
+
+          if (resolver != null) {
+            resolverSet.add(parameterName, resolver);
```

Comment:
Did this change but it didn't make the code better. getResolver needs to receive parameters, parameterModel and the resolved member name. After doing that, the only relevant change is that instead of the if there's an ifPresent. I don't see any improvement in legibility, on the contrary